rllib/env/tests/test_single_agent_episode.py

```diff
@@ -0,0 +1,303 @@
+import gymnasium as gym
+import numpy as np
+import unittest
+
+from gymnasium.core import ActType, ObsType
+from typing import Any, Dict, Optional, SupportsFloat, Tuple
+
+import ray
+from ray.rllib.env.single_agent_episode import SingleAgentEpisode
+
+# TODO (simon): Add to the tests `info` and `extra_model_outputs`
+# as soon as #39732 is merged.
+
+
+class TestEnv(gym.Env):
+    def __init__(self):
+        self.observation_space = gym.spaces.Discrete(201)
+        self.action_space = gym.spaces.Discrete(200)
+        self.t = 0
+
+    def reset(
+        self, *, seed: Optional[int] = None, options=Optional[Dict[str, Any]]
+    ) -> Tuple[ObsType, Dict[str, Any]]:
+        self.t = 0
+        return 0, {}
+
+    def step(
+        self, action: ActType
+    ) -> Tuple[ObsType, SupportsFloat, bool, bool, Dict[str, Any]]:
+        self.t += 1
+        if self.t == 200:
+            is_terminated = True
+        else:
+            is_terminated = False
+
+        return self.t, self.t, is_terminated, False, {}
+
+
+class TestSingelAgentEpisode(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        ray.shutdown()
+
+    def test_init(self):
+        # Create empty episode.
+        episode = SingleAgentEpisode()
+        # Empty episode should have a start point and count of zero.
+        self.assertTrue(episode.t_started == episode.t == 0)
+
+        # Create an episode with a specific starting point.
+        episode = SingleAgentEpisode(t_started=10)
+        self.assertTrue(episode.t == episode.t_started == 10)
+
+        # Sample 100 values and initialize episode with observations.
+        env = gym.make("CartPole-v1")
+        observations = rewards = is_terminateds = is_truncateds = []
+        init_obs, _ = env.reset()
+        observations.append(init_obs)
+        for _ in range(100):
+            action = env.action_space.sample()
+            obs, reward, is_terminated, is_truncated, _ = env.step(action)
```

Comment:
Can we add `infos` to this test as well?

Reply:
As soon as the ENvRunner PR is merged and can merge the master into this PR and add `infos` and `extra_model_outputs`.